packages/shared/lib/sdk/sync.ts

```diff
@@ -482,7 +488,40 @@ export class NangoAction {
         }
 
         if (this.dryRun) {
-            return this.nango.proxy(config);
+            const proxyResponse = await this.nango.proxy(config);
+            if (this.saveResponses) {
+                const directoryName = `${process.env['NANGO_MOCKS_RESPONSE_DIRECTORY']}${config.providerConfigKey}`;
+                if (directoryName) {
+                    const fileName = config.endpoint.replace('/', '');
+                    if (!fs.existsSync(directoryName)) {
+                        fs.mkdirSync(directoryName);
+                        fs.mkdirSync(`${directoryName}/mocks`);
+                    }
+                    let filePath = `${directoryName}/mocks/${fileName}.json`;
+                    if (config.paginate) {
+                        const paginateType = config.method.toLowerCase();
+                        if (!fs.existsSync(`${directoryName}/mocks/paginate/${paginateType}`)) {
+                            fs.mkdirSync(`${directoryName}/mocks/paginate/${paginateType}`, { recursive: true });
+                        }
+                        if (!fs.existsSync(`${directoryName}/mocks/paginate/${paginateType}/${this.syncConfig?.sync_name}`)) {
+                            fs.mkdirSync(`${directoryName}/mocks/paginate/${paginateType}/${this.syncConfig?.sync_name}`, { recursive: true });
+                        }
+                        if (fileName.includes('/')) {
+                            const fileNameParts = fileName.split('/');
+                            fileNameParts.pop();
+                            const additionalPath = fileNameParts.join('/');
+                            fs.mkdirSync(`${directoryName}/mocks/paginate/${paginateType}/${this.syncConfig?.sync_name}/${additionalPath}`, {
+                                recursive: true
+                            });
+                        }
+                        filePath = `${directoryName}/mocks/paginate/${paginateType}/${this.syncConfig?.sync_name}/${fileName}.json`;
+                    }
+                    const responsePath = config.paginate?.response_path;
+                    const paginateData = responsePath ? proxyResponse.data[responsePath] : proxyResponse.data;
+                    fs.writeFileSync(filePath, JSON.stringify(paginateData, null, 2));
+                }
+            }
+            return proxyResponse;
```

Comment:
It's probably handy to do it here, but mixing that up in the SDK is a bit annoying. 
Is there any way to decouple this with events or by storing it in memory until further processing?

Reply:
Thanks, updated